Assets/HoloToolkit/Input/Scripts/Voice/SpeechInputSource.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using UnityEngine.Windows.Speech;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// SpeechInputSource allows you to specify keywords and methods in the Unity
+    /// Inspector, instead of registering them explicitly in code.
+    /// This also includes a setting to either automatically start the
+    /// keyword recognizer or allow your code to start it.
+    ///
+    /// IMPORTANT: Please make sure to add the microphone capability in your app, in Unity under
+    /// Edit -> Project Settings -> Player -> Settings for Windows Store -> Publishing Settings -> Capabilities
+    /// or in your Visual Studio Package.appxmanifest capabilities.
+    /// </summary>
+    public partial class SpeechInputSource : BaseInputSource
```

Comment:
@HodgsonSDAS In the new InputManager architecture, event sources register with it. It then looks for a message handler in the object with the focus. If not there, it goes up the object hierarchy. It manager also allows global handlers and fallback handlers.
The SpeechInputSource is only the source that could replace the KeywordManager. I currently only support handlers for when the keyword is spoken as scripts. I'm prototyping support for inspector. 
If this PR results in a better solution than the KeywordsManager, it would be nice to turn it obsolete as having both can be confusing to users.
It's also my fault but I've been trying to clear all reviews and this PR is loosing some focus.